runtime-modules/council/src/lib.rs

```diff
@@ -0,0 +1,704 @@
+//! - [release_candidacy_stake](./struct.Module.html#method.release_candidacy_stake)
+//!
+//! ## Important functions
+//! These functions have to be called by the runtime for the council to work properly.
+//! - [recieve_referendum_results](./trait.ReferendumConnection.html#method.recieve_referendum_results)
+//! - [can_release_vote_stake](./trait.ReferendumConnection.html#method.can_release_vote_stake)
+//!
+//! ## Dependencies:
+//! - [proposals engine](../referendum/index.html)
+//!
+//! NOTE: When implementing runtime for this module, don't forget to call all ReferendumConnection
+//!       trait functions at proper places.
+
+/////////////////// Configuration //////////////////////////////////////////////
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// used dependencies
+use codec::{Codec, Decode, Encode};
+use frame_support::traits::{Currency, Get};
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage, ensure, error::BadOrigin, Parameter,
+};
+
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_arithmetic::traits::BaseArithmetic;
+use sp_runtime::traits::{MaybeSerialize, Member};
+use std::marker::PhantomData;
+use system::{ensure_signed, RawOrigin};
+
+use referendum::Instance as ReferendumInstanceGeneric;
+use referendum::Trait as ReferendumTrait;
+use referendum::{OptionResult, ReferendumManager};
+
+// declared modules
+mod mock;
+mod staking_handler;
+mod tests;
+
+use staking_handler::StakingHandler2;
+
+/////////////////// Data Structures ////////////////////////////////////////////
+
+/// Information about council's current state and when it changed the last time.
+#[derive(Encode, Decode, PartialEq, Eq, Debug, Default)]
+pub struct CouncilStageUpdate<BlockNumber> {
+    stage: CouncilStage,
+    changed_at: BlockNumber,
+}
+
+/// Possible council states.
+#[derive(Encode, Decode, PartialEq, Eq, Debug)]
+pub enum CouncilStage {
+    /// Candidacy announcement period.
+    Announcing(CouncilStageAnnouncing),
+    /// Election of the new council.
+    Election(CouncilStageElection),
+    /// The idle phase - no new council election is running now.
+    Idle,
+}
+
+impl Default for CouncilStage {
+    fn default() -> CouncilStage {
+        CouncilStage::Announcing(CouncilStageAnnouncing {
+            candidates_count: 0,
+        })
+    }
+}
+
+/// Representation for announcing candidacy stage state.
+#[derive(Encode, Decode, PartialEq, Eq, Debug, Default)]
+pub struct CouncilStageAnnouncing {
+    candidates_count: u64,
+}
+
+/// Representation for new council members election stage state.
+#[derive(Encode, Decode, PartialEq, Eq, Debug, Default)]
+pub struct CouncilStageElection {
+    candidates_count: u64,
+}
+
+/// Candidate representation.
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Encode, Decode, PartialEq, Eq, Debug, Default, Clone)]
+pub struct Candidate<AccountId, Balance> {
+    account_id: AccountId,
```

Comment:
We identify candidates with a member, not an account.

Reply:
We actually identification Candidates (and elected members) by `council_user_id`, but we need to know `account_id` as well so we can use it for staking a currency. It is necessary for an account never to switch `council_user_id` id in the runtime, as we discussed. I've added commentary about this to the module's description.